pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
 	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
 
+	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
+		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
```

Comment:
I guess.  I'm suddenly skeptical about defaulting based on detected docker
in something that is intended to be set by admins.  Why is this something
we are doing in our code, and not simply a config flag?

On Mon, Feb 1, 2016 at 9:43 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/cmd/server/kubernetes/node_config.go
> <https://github.com/openshift/origin/pull/6538#discussion_r51515757>:
>
> > @@ -148,6 +152,12 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
> >  	server.PodInfraContainerImage = imageTemplate.ExpandOrDie("pod")
> >  	server.CPUCFSQuota = true // enable cpu cfs quota enforcement by default
> >
> > +	if dockerClient, _, err := docker.NewHelper().GetClient(); err == nil {
> > +		if docker.IsAPIVersionCompatible(dockerClient, minimumDockerAPIVersionForParallelPulls) {
>
> Would it? This only sets serial to false if it's able to get the docker
> client and the version is ok. If it encounters an error, it keeps the
> default serial=true and moves on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6538/files#r51515757>.
>